File: app/src/main/java/revisao/ActRevisaoFrags.java
package revisao;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.exemplo.R;

public class ActRevisaoFrags extends AppCompatActivity {

    TextView benVindo;
    Button vaParaFragmentos, vaParaComunicaFrags;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.act_main_revisao);
        Bundle extras = getIntent().getExtras();
        String nome = extras != null ? extras.getString("nome") : null;

        benVindo = findViewById(R.id.benvindo_nome);
        String textoFormatado = "Bem vindo " + nome + "! Clique abaixo para ver Fragmentos:";
        benVindo.setText(textoFormatado);

        vaParaFragmentos = findViewById(R.id.benvindo_button);
        vaParaComunicaFrags = findViewById(R.id.benvindo_button_comunica_frags);

        vaParaFragmentos.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(v.getContext(), ActFragmentsRevisao.class));
            }
        });

        vaParaComunicaFrags.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(v.getContext(), ActComunicaFragsRevisao.class));
            }
        });

    }
}

File: app/src/main/java/revisao/ActRevisaoFrags.kt
package revisao

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import android.os.Bundle
import com.example.exemplo.R
import android.content.Intent
import android.view.View
import kotlin.reflect.KClass

class ActRevisaoFrags : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.act_main_revisao)

        val textoFormatado =
            "Bem vindo(a) ${intent.extras?.getString("nome") ?: "Gunther"}! Clique abaixo para ver Fragmentos:"

        findViewById<TextView>(R.id.benvindo_nome).text = textoFormatado

        findViewById<View>(R.id.benvindo_button).setOnClickListener { view ->
            view.start(ActFragmentsRevisao::class)
        }

        findViewById<View>(R.id.benvindo_button_comunica_frags).setOnClickListener {
            it.start(ActComunicaFragsRevisao::class)
        }
    }

    private fun <T : Activity> View.start(klass: KClass<T>) =
        startActivity(Intent(context, klass.java))
}